plugin/evm/ftso_system_test.go

```diff
@@ -0,0 +1,221 @@
+	testAbiSubmitterBin  = `0x608060405234801561001057600080fd5b506040516103ee3803806103ee8339818101604052606081101561003357600080fd5b81019080805190602001909291908051906020019092919080519060200190929190505050826000806101000a81548173ffffffffffffffffffffffffffffffffffffffff021916908373ffffffffffffffffffffffffffffffffffffffff16021790555081600160006101000a81548173ffffffffffffffffffffffffffffffffffffffff021916908373ffffffffffffffffffffffffffffffffffffffff16021790555080600260006101000a81548173ffffffffffffffffffffffffffffffffffffffff021916908373ffffffffffffffffffffffffffffffffffffffff1602179055505050506102c28061012c6000396000f3fe608060405234801561001057600080fd5b50600436106100625760003560e01c806311a7aaaa1461006757806338b5f8691461009b57806371e1fad9146100cf5780638c9d28b614610103578063b39c685814610137578063c2b0d47b1461016b575b600080fd5b61006f61019f565b604051808273ffffffffffffffffffffffffffffffffffffffff16815260200191505060405180910390f35b6100a36101c5565b604051808273ffffffffffffffffffffffffffffffffffffffff16815260200191505060405180910390f35b6100d76101eb565b604051808273ffffffffffffffffffffffffffffffffffffffff16815260200191505060405180910390f35b61010b610214565b604051808273ffffffffffffffffffffffffffffffffffffffff16815260200191505060405180910390f35b61013f61023e565b604051808273ffffffffffffffffffffffffffffffffffffffff16815260200191505060405180910390f35b610173610268565b604051808273ffffffffffffffffffffffffffffffffffffffff16815260200191505060405180910390f35b600260009054906101000a900473ffffffffffffffffffffffffffffffffffffffff1681565b600160009054906101000a900473ffffffffffffffffffffffffffffffffffffffff1681565b60008060009054906101000a900473ffffffffffffffffffffffffffffffffffffffff16905090565b6000600160009054906101000a900473ffffffffffffffffffffffffffffffffffffffff16905090565b6000600260009054906101000a900473ffffffffffffffffffffffffffffffffffffffff16905090565b60008054906101000a900473ffffffffffffffffffffffffffffffffffffffff168156fea264697066735822122035d0b8aa6a058839885934e180e1ddb5af32130473692bb78427ccd544f1323f64736f6c63430007060033`
+	testAbiSubmitter, _  = abi.JSON(strings.NewReader(testAbiSubmitterJSON))
+)
+
+func TestFTSOSystem_Contracts(t *testing.T) {
+	t.Run("common case", func(t *testing.T) {
+		auth, be := simulatedBlockchain(t)
+
+		// VotePower contract
+		vpInt := big.NewInt(100)
+		votePowerAddr := deployTestContract(t, auth, be, testAbiVotePower, testAbiVotePowerBin, vpInt)
+
+		// Wnat contract
+		wnatAddr := deployTestContract(t, auth, be, testAbiWnat, testAbiWnatBin, votePowerAddr)
+
+		// Reward contract
+		unclaimedReward := big.NewInt(100)
+		rewardAddr := deployTestContract(t, auth, be, testAbiReward, testAbiRewardBin, wnatAddr, unclaimedReward)
+
+		// Manager contract
+		rewardEpochDurationSeconds := big.NewInt(100)
+		rewardEpochPowerHeight := big.NewInt(100)
+		rewardEpochStartHeight := big.NewInt(100)
+		rewardEpochStartTime := big.NewInt(100)
```

Comment:
Same here. These all need to be unique values, so we are sure we are distinguishing them correctly when retrieving.